modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -17,61 +17,125 @@
 namespace cv {
 namespace gapi {
 
+/**
+* \addtogroup gapi_serialization
+* @{
+*/
+
 namespace detail {
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS cv::GComputation getGraph(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS cv::GMetaArgs getMetaArgs(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS cv::GRunArgs getRunArgs(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS std::vector<std::string> getVectorOfStrings(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     template<typename... Types>
     cv::GCompileArgs getCompileArgs(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     template<typename RMatAdapterType>
     cv::GRunArgs getRunArgsWithRMats(const std::vector<char> &p);
 } // namespace detail
 
+/** @brief This function allows to serialize GComputation.
+ *
+ * Check different overloads for more examples.
+ * @param c GComputation to serialize.
+ * @return serialized vector of bytes.
+ */
 GAPI_EXPORTS std::vector<char> serialize(const cv::GComputation &c);
-//namespace{
 
-template<typename T> static inline
-T deserialize(const std::vector<char> &p);
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const cv::GCompileArgs& ca);
```

Comment:
Please document what the parameter is

Reply:
(here and below)

Reply:
Note sure why, but done

Reply:
That's the key thing about differentiating the overloads